torch/nn/quantized/modules/conv.py

```diff
@@ -45,8 +45,12 @@ def __init__(self, in_channels, out_channels, kernel_size, stride,
         self.groups = groups
         self.padding_mode = padding_mode
         # Initialize as NCHW. set_weight will internally transpose to NHWC.
+        if self.transposed:
+            weight_shape = [in_channels, out_channels // self.groups]
```

Comment:
I suspect something wrong here. Are you sure dimensions are supposed to be like this. I see below that `bias_float`'s size is out_channels. It should also depend on transposed param? So here : https://github.com/pytorch/pytorch/pull/40370/files#diff-d0713fbda560de6aecef93e4c3d973a5R561, you are providing #of output channels differently than what your weight is telling you.
But more importantly what is going on with I dont understand why in deconv weigth's #of output channels is equal to in_channel? I would have thought it should not change right? Your deconv op's weight should tell you how many output channels you want to produce and that should not depend on # of input channels. No? Only thing different here I imagine is output shape calculation.

Reply:
bias size should always be equal to the number of output channels. The shape of the weight matrix in convolution and transposed convolution is different:

- convolution weight shape: [out_channels, in_channels / groups, k0, k1, k2, ...]
- convolution transpose weight shape: [in_channels, out_channels / groups, k0, k1, k2, ...]

Can you elaborate what you mean when you say "deconv weights #of output channels equals to in_channels"?

Reply:
Thanks Kimish for finding the bug!